irohad/network/impl/peer_tls_certificates_provider.hpp

```diff
@@ -0,0 +1,43 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef IROHA_PEER_TLS_CERTIFICATES_PROVIDER_HPP
```

Comment:
This header defines an interface and at the same time is placed under `.../impl/...`.
Looks like the file can be moved out of the `impl` directory.